Sources/BraintreePayPal/BTPayPalApprovalURLParser.swift

```diff
@@ -17,31 +17,41 @@ enum PayPalRedirectType {
 /// Parses response body from `/v1/paypal_hermes/*` POST requests to determine the `PayPalRedirectType`
 struct BTPayPalApprovalURLParser {
     
-    var redirectType: PayPalRedirectType
-    
-    var pairingID: String? {
-        switch redirectType {
-        case .webBrowser(let url), .payPalApp(let url):
-            let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: true)?
-                .queryItems?
-                .compactMap { $0 }
-
-            if let baToken = queryItems?.filter({ $0.name == "ba_token" }).first?.value, !baToken.isEmpty {
-                return baToken
-            } else if let ecToken = queryItems?.filter({ $0.name == "token" }).first?.value, !ecToken.isEmpty {
-                return ecToken
-            }
+    let redirectType: PayPalRedirectType
+    let url: URL
 
-            return nil
+    var ecToken: String? {
```

Comment:
Would we want to have some test coverage on the logic in this getter?

Reply:
We don't actually have any tests on this struct as a whole. It's implicitly tested through the client. Thoughts on making a follow up ticket for this work so we can add a test class for this file?

Reply:
Gotcha. I'm not sure how the iOS testing framework is set up but I think having tests cover smaller scopes makes things easier to understand and maintain.

`BTPayPalApprovalURLParser` looks like it could be a dependency that can be mocked for the client unit tests. This way it'd be easy to return expected `ecToken` values for client tests. And the logic within the parser itself can be completely separate tests.

Completely flexible on how we want to handle it! (creating a follow-up ticket or maybe getting team alignment on how we want to approach testing)

Reply:
Oh sorry if I was not clear, yes we should _absolutely_ have tests for `BTPayPalApprovalURLParser` - since we have 0 now, it expands the scope of this PR to add them all here. I created DTBTSDK-3804 to address tests as a whole for this file.

Reply:
Ah got it! Thanks!